XamarinCommunityToolkit/Views/AvatarView/ImageSourceValidator.android.ios.macos.uwp.wpf.gtk.tizen.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+#if MONOANDROID
+using Xamarin.Forms.Platform.Android;
+#elif __IOS__
+using Xamarin.Forms.Platform.iOS;
+#elif __MACOS__
+using Xamarin.Forms.Platform.MacOS;
+#elif UWP
+using Xamarin.Forms.Platform.UWP;
+#elif NET471
+using Xamarin.Forms.Platform.GTK.Renderers;
+#elif TIZEN
+using Xamarin.Forms.Platform.Tizen;
+#else
+using Xamarin.Forms.Platform.WPF;
+#endif
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+#if MONOANDROID || __IOS__ || __MACOS__
+	using UriImageSourceHandler = ImageLoaderSourceHandler;
+#endif
+
+#if MONOANDROID || __IOS__ || __MACOS__ || NET471
+	using StreamImageSourceHandler = StreamImagesourceHandler;
+#endif
+
+	class ImageSourceValidator : IImageSourceValidator
+	{
+		public async Task<bool> IsImageSourceValidAsync(ImageSource source)
+		{
+			var handler = GetHandler(source);
+
```

Comment:
need to insert a check

```csharp
if (handler == null)
    return false;
```

Reply:
That solves for me

Reply:
if we return False the image won't displayed if ImageSource is GravatarImageSource. Is it expected?

Reply:
@VladislavAntonyuk I tried here, and the image displayed after loading. So maybe it's null when we load the page in first time. At least for android

Reply:
Hm, you are right. Then let's return True

Reply:
@pictos AvatarView line 185. if we return false the image IsVisible should be false.

Reply:
so... Why return false works?

Reply:
@AndreiMisiukevich we should return false. Otherwise Label is not visible, because Label.IsVisible=!Image.IsVisible

Reply:
I added more ImageSources (GravatarImageSource, FontImageSource)